tests/conftest.py

```diff
@@ -1,15 +1,62 @@
 import os
+from typing import AsyncGenerator
 
+import httpx
 import pytest
+import pytest_asyncio
+from asgi_lifespan import LifespanManager
+from google.cloud.aiplatform.constants.base import DEFAULT_REGION
+from httpx import ASGITransport
+from openai import AsyncAzureOpenAI
 
-from tests.server import server_generator
 
-DEFAULT_API_VERSION = "2023-03-15-preview"
-TEST_SERVER_URL = os.getenv("TEST_SERVER_URL", "http://0.0.0.0:5001")
+@pytest.fixture(autouse=True)
+def configure_unit_tests(monkeypatch, request):
+    """
+    Set up fake environment variables for unit tests.
+    """
+    if "tests/unit_tests" in request.node.nodeid:
+        monkeypatch.setenv("GOOGLE_APPLICATION_CREDENTIALS", "test-creds")
+        monkeypatch.setenv("DEFAULT_REGION", DEFAULT_REGION)
+        monkeypatch.setenv("GCP_PROJECT_ID", "test-project-id")
 
 
-@pytest.fixture(scope="module")
-def server():
-    yield from server_generator(
-        "aidial_adapter_vertexai.app:app", TEST_SERVER_URL
-    )
+@pytest.fixture(autouse=True)
+def disable_aiocache():
+    # It's important to disable caches defined in aidial_adapter_vertexai.vertex_ai,
+    # because pytest recreates app for every test function,
+    # but the caches are shared across the tests.
+    #
+    # A cached model holds a reference to an event loop
+    # that is created for the first app instance.
+    # Once the first app instance is released, the event loop is closed.
+    # This reference to a closed event loop breaks follow-up tests.
+    os.environ["AIOCACHE_DISABLE"] = "1"
+
+
+@pytest_asyncio.fixture()
```

Comment:
Could you pls explain, why you can't just use
```python

@pytest.fixture
async def test_http_client() ...
```
?

Reply:
Fixed by switching to [auto mode](https://pypi.org/project/pytest-asyncio/0.20.1/)